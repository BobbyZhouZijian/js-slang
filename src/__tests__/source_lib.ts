import { runInContext } from '../index'
import { mockContext } from '../mocks/context'
import { Finished } from '../types'

test('Can run loaded source lib', () => {
  const code = 'factorial(5);'
  const context = mockContext(2)
  const promise = runInContext(code, context, { scheduler: 'preemptive' })
  return promise.then(obj => {
    expect(obj).toMatchSnapshot()
    expect(context.errors).toMatchSnapshot()
    expect(obj.status).toBe('finished')
    expect((obj as Finished).value).toBe(120)
  })
})
