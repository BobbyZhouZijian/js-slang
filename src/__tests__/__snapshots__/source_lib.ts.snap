// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can run loaded source lib 1`] = `
Object {
  "status": "finished",
  "value": 120,
}
`;

exports[`Can run loaded source lib 2`] = `Array []`;
